### DISCUSSION TITLE
Living without appc - errors again

### DISCUSSION THREAD

**Discussion by RomanLysiukOSF** [appcelerator/titanium_mobile#13312] [Q&A]
In additions to the discussion #13310 
On another Macbook with identical configuration we managed to compile the same application but it throws an error into console right after start. At the beginning it should display modal window and probably this modal fails:
![image](https://user-images.githubusercontent.com/31540999/158426854-6399d18b-bfd7-4586-9bd1-87b07cf2657e.png)

Error text:
```
[ERROR] Script Error: {
[ERROR]     column = 30;
[ERROR]     line = 209;
[ERROR]     message = "invalid method (createMatrix2D) passed to UIModule";
[ERROR]     nativeStack = "3   CoreFoundation  0x0000000111ea403c _invoking__ + 140\n4   CoreFoundation  0x0000000111ea14d5 -[NSInvocation invoke] + 325\n5   EndlessAisle MO PT DEV v1.12   0x000000010a45348c -[KrollMethod call:] + 748\n6   EndlessAisle MO PT DEV v1.12   0x000000010a452252 _KrollCallAsFunction_block_invoke + 50\n7   EndlessAisle MO PT DEV v1.12   0x000000010a56799e TiThreadPerformOnMainThread + 78\n8   EndlessAisle MO PT DEV v1.12   0x000000010a451f00 KrollCallAsFunction + 704\n9   JavaScriptCore  0x000000010ad7b5b3 ZN3JSC16JSCallbackObjectINS_20JSDestructibleObjectEE4callEPNS_9ExecStateE + 515\n10  JavaScriptCore  0x000000010b309571 ZN3JSC5LLInt9setUpCallEPNS_9ExecStateEPNS_11InstructionENS_22CodeSpecializationKindENS_7JSValueEPNS_17LLIntCallLinkInfoE + 385\n11  JavaScriptCore  0x000000010ad4832f llint_entry + 26803\n12  JavaScriptCore  0x000000010ad4833d llint_entry + 26817\n13  JavaScriptCore  0x000000010ad418fe vmEntryToJavaScript + 244\n14  JavaScriptCore  0x000000010b23d846 ZN3JSC11Interpreter11executeCallEPNS_9ExecStateEPNS_8JSObjectENS_8CallTypeERKNS_8CallDataENS_7JSValueERKNS_7ArgListE + 470\n15  JavaScriptCore  0x000000010b43e9b4 _ZN3JSC12profiledCallEPNS_9ExecStateENS_15ProfilingReasonENS_7JSValueENS_8CallTypeERKNS_8CallDataES3_RKNS_7ArgListE + 196\n16  JavaScriptCore  0x000000010ad812d4 JSObjectCallAsFunction + 468\n17  EndlessAisle MO PT DEV v1.12   0x000000010a44b8d7 -[KrollCallback call:thisObject:] + 727\n18  EndlessAisle MO PT DEV v1.12   0x000000010a44ccd0 -[KrollEvent invoke:] + 272\n19  EndlessAisle MO PT DEV v1.12   0x000000010a44d303 -[KrollContext invoke:] + 163";
[ERROR]     sourceURL = "file:///Users/admin/Library/Developer/CoreSimulator/Devices/91C1614B-3177-4CBB-8334-1F44294D1BEB/data/Containers/Bundle/Application/3B872E51-56B7-48CD-97E6-5116ABBFC7A0/EndlessAisle%20MO%20PT%20DEV%20v1.12.app/alloy/animation.js";
[ERROR]     stack = "    at [native code]\n    at mailto:popin@file:///users/admin/library/developer/coresimulator/devices/91c1614b-3177-4cbb-8334-1f44294d1beb/data/containers/bundle/application/3b872e51-56b7-48cd-97e6-5116abbfc7a0/endlessaisle%20mo%20pt%20dev%20v1.12.app/alloy/animation.js:209:30)\n    at file:///Users/admin/Library/Developer/CoreSimulator/Devices/91C1614B-3177-4CBB-8334-1F44294D1BEB/data/Containers/Bundle/Application/3B872E51-56B7-48CD-97E6-5116ABBFC7A0/EndlessAisle%20MO%20PT%20DEV%20v1.12.app/alloy/controllers/index.js:403:22)";
[ERROR]     toJSON = "<KrollCallback: 0x600001d50200>";
[ERROR] }
```

`invalid method (createMatrix2D) passed to UIModule` - how this error could appear after switching from `appc` to `ti`?

Thanks,
Roman

**Comment by m1ga** [comment]
It probably didn't appear by switching the CLI but by using newer versions of the toolchain like Alloy. And did you update the SDK too? 

Things you should do:
* search for createMatrix2D and see if there is anything wrong
* check something with `mailto:popin` :shrug: That is strange in the error log 
* check `app/alloy/animation.js:209:30` and `controllers/index.js:403:22`

**Comment by RomanLysiukOSF** [comment]
Hello @m1ga 
1. "createMatrix2D" is not mentioned in our project
2. "mailto:popin" also not a part of project code
3. File `app/alloy/animation.js` doesn't exist at all.
4. File `controllers/index.js` exists but line 403 is `failStartup(_L('Unable to login kiosk user.'), null, true);` where `failStartup` is a simple function presents a dialog to the user, but I don't understand why this function is not mentioned in the error log

Thanks,
Roman

**Reply by m1ga** [reply]
sorry for the confusion 😄 the alloy/animation file is a build-in file. And the animation that is triggert is:
https://github.com/tidev/alloy/blob/5e9e25e58ce0fac3e8c12e7583297a5f64d27101/Alloy/builtins/animation.js#L200
so something like 
https://github.com/tidev/alloy/blob/5e9e25e58ce0fac3e8c12e7583297a5f64d27101/test/apps/testing/ALOY-632/controllers/index.js#L3

**Reply by m1ga** [reply]
ah, and since you are using such an old SDK it might need to be the old `create2DMatrix ` instead of `createMatrix2D`. But that is inside alloy. So you might need to install an older Alloy if you installed a newer version (or update the SDK :) )
